pkg/didcomm/protocol/introduce/metadata.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package introduce
+
+import (
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/didexchange"
+)
+
+const (
+	metaContextID    = "context_id"
+	metaSkipProposal = "skip_proposal"
+	metaInvitation   = "invitation"
+	metaRecipients   = "recipients"
```

Comment:
When we start putting more real data in the messenger's storage space, I start wondering about data lifecycle ... and if we should be namespacing according to the protocol (aka in the protocol's storage space).

Reply:
Added prefix!

Reply:
Actually my point was more about should the service be storing (some of) these fields to its own storage provider.

Reply:
Currently, the messenger stores and provides it. If service stores it than storing metadata in the messenger is redundant.
The actual logic is that message metadata is stored under the thread-ID key. And metadata is automatically populated when a new message comes (with the same thread). Metadata is only available for one agent (who provided metadata). Actually,  metadata is a thread based key-value storage. Only one record is being stored per thread.

`Should the service be storing some of these fields to its own storage provider` - I do not see the reason for that. Only if we remove this common metadata logic. About protocol namespace,  metadata is thread-base storage. And the thread can't be duplicated. Looks like we do not need this namespace. What we can do it is to open a new store for metadata. But even for that no reason.